### ISSUE TITLE
support for new biome data

### ISSUE THREAD

**Issue by cabbey** [bedrock-viz/bedrock-viz#172]
In 1.18 mojang introduced a new way to store biome data. We need to parse it out of the new records instead of just blindly taking the data bit we were using before. Will need to figure out a trigger though, because some old chunks that is still the right data. Additionally  there appear to be at least 3 different layers of biome: surface, underground, deep underground... and some layer blending going on between them. We need to investigate what the data tells us and how we should present it to the user.

**Comment by geoffholden** [comment]
I took a crack at this a while back and had a quick & dirty parser in place for it that only kept the surface biome. I stopped working on it because @Desterly said he had a proper parser in place for it, but I still have it on one of my branches.

As for the data, the biome data is stored in a new record, so reading it vs old chunks is straightforward. But the biome is now stored per block, so you could (in theory) have a different biome at every single level change for every single block in the chunk.

I think we have a couple of different options for displaying the new data:
- Generate biome images for every single layer--slow and probably overkill
- Pick a couple of standard levels to generate images for (surface, cave, deep dark?)
- Split the difference somewhere and generate images for every _n_ layers (16 or 32, maybe?)
These could all be implemented via command line options as well.

In the interest of having something working in the sort term, we could get a parser in place for just the surface biome--that wouldn't require any frontend work and while it wouldn't show the new underground biomes, it would work like it did for older versions.

**Comment by cabbey** [comment]
yeah, I was thinking it would be interesting to capture every layer and just see if there were any natural patterns that stood out to inform us what would be useful. I expect the three we both named are the extent of what we really need to give users.

Something short term would probably be a good start, especially if @scp-r wanted to try to get a new release out mid this month, as it would at least get rid of the random noise in the current biome map.

**Comment by geoffholden** [comment]
You can see what I had put together [here](/geoffholden/bedrock-viz/tree/biome-3d). It's a bit behind master, but I think it'd merge cleanly if you want to pull it in somewhere to take a look.
